### PR TITLE
GRF-339: Apply category permissions on children after saving

### PR DESCRIPTION
**Pull request by BryanS29-akeneo**
**Description (for Contributor and Core Developer)**

- Add a new `CategoryEditedEvent` dispatched in the `UpdateCategoryController`
- Creation of a new sql query to retrieve category children ids

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by jeremyBeaucousinAkeneo — APPROVED** [review]

File: src/Akeneo/Category/back/Application/Query/GetCategoryChildrenIdsInterface.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+declare(strict_types=1);
+
+namespace Akeneo\Category\Application\Query;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface GetCategoryChildrenIdsInterface
```
Comment (etiennekerbastard) [inline src/Akeneo/Category/back/Application/Query/GetCategoryChildrenIdsInterface.php:11]: In DDD context, adding Interface in the name of the interface does not seem right as we are adding a technical term in the application layer
```suggestion
interface GetCategoryChildrenIds
```

**Review by olivierpontier — APPROVED** [review]

File: src/Akeneo/Category/back/Api/Event/CategoryEditedEvent.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+namespace Akeneo\Category\Api\Event;
+
+use Akeneo\Category\Api\Command\UserIntents\UserIntent;
+use Akeneo\Category\Domain\Model\Enrichment\Category;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class CategoryEditedEvent
```
Comment (etiennekerbastard) [inline src/Akeneo/Category/back/Api/Event/CategoryEditedEvent.php:14]:
```suggestion
class CategoryUpdatedEvent
```

**Review by etiennekerbastard — APPROVED** [review]

File: src/Akeneo/Category/back/Api/Event/CategoryEditedEvent.php (hunk 2 above)
Comment (etiennekerbastard) [reply src/Akeneo/Category/back/Api/Event/CategoryEditedEvent.php:14]: nevermind

Comment (BryanS29-akeneo) [reply src/Akeneo/Category/back/Api/Event/CategoryEditedEvent.php:14]: not the same event, we are in the edit phase of the category (front => back controller), the update event will be in the `UpsertCategoryCommandHandler` when we have already set our Category model

**Pull request merged**